polkadot/node/primitives/src/lib.rs

```diff
@@ -524,6 +524,8 @@ pub struct SubmitCollationParams {
 	/// okay to just drop it. However, if it is called, it should be called with the signed
 	/// statement of a parachain validator seconding the collation.
 	pub result_sender: Option<futures::channel::oneshot::Sender<CollationSecondedSignal>>,
+	/// The core index on which the resulting candidate should be backed
+	pub core_index: CoreIndex,
```

Comment:
Is this a breaking change for parachains? If so, can we avoid it?

Reply:
I guess so, we can avoid it by introducing `SubmitCollationParams2` or something like that. Maybe an enum for future proofing.

Reply:
is it a breaking change though? it's not part of a network protocol, just an overseer message.

Reply:
What @eskimor suggests is to not force collators to upgrade if they don't use elastic scaling. The `SubmitCollationParams` has been altered so if they would pull latest polkadot sdk , they will have to also change the node side code. I also don't think there is much impact, because they are likely using the same unaltered node side code, but we should follow best practice and avoid this scenario as well.

Reply:
Oh well, forgot now we have a monorepo, they’ll have to fetch everything so I guess that is fine unless they are actually building their collator nodes separately.

Reply:
Indeed. I am asking for source level changes. If this is not user facing for 99% this is good. Definitely worth mentioning in the release notes though + howto upgrade for the remaining 1%?

Reply:
I'll add that missing PRDoc